DependencyInjection/Compiler/EntityListenerPass.php

```diff
@@ -52,7 +52,12 @@ public function process(ContainerBuilder $container)
                     $this->attachToListener($container, $name, $id, $attributes);
                 }
 
-                $container->getDefinition($resolver)->addMethodCall('register', array(new Reference($id)));
+                if (isset($attributes['lazy']) && $attributes['lazy']) {
+                    $listener = $container->getDefinition($id);
+                    $container->getDefinition($resolver)->addMethodCall('registerService', array($listener->getClass(), $id));
```

Comment:
This method is not in the interface so it will break for people tho changes the resolver.